backend/src/services/worker/worker.service.ts

```diff
@@ -0,0 +1,58 @@
+import { worker as workerRep } from '~/data/repositories/repositories';
+import {
+  EAMWorkerCreateRequestDto,
+  EAMWorkerResponseDto,
+} from '~/common/types/types';
+import { Worker as WorkerEntity } from './worker.entity';
+import { getRandomId as getRandomName } from '../../../../shared/build';
+import {
+  encrypt as encryptServ,
+  tenant as tenantServ,
+} from '~/services/services';
+
+type Constructor = {
+  workerRepository: typeof workerRep;
+  tenantService: typeof tenantServ;
+  encryptService: typeof encryptServ;
+};
+
+class Worker {
+  #workerRepository: typeof workerRep;
+  #tenantService: typeof tenantServ;
+  #encryptService: typeof encryptServ;
+
+  constructor({
+    workerRepository,
+    encryptService,
+    tenantService,
+  }: Constructor) {
+    this.#workerRepository = workerRepository;
+    this.#tenantService = tenantService;
+    this.#encryptService = encryptService;
+  }
+
+  public async create({
+    name,
+    password,
+  }: EAMWorkerCreateRequestDto): Promise<EAMWorkerResponseDto> {
+    const tenant = await this.#tenantService.create(getRandomName());
```

Comment:
We need to have workers connected to an existing tenants, not to a new ones

Reply:
working on it